server/src/main/java/org/elasticsearch/action/termvectors/TransportTermVectorsAction.java

```diff
@@ -69,21 +76,18 @@ protected ShardIterator shards(ProjectState project, InternalRequest request) {
                 .getFirst();
         }
 
-        return operationRouting.useOnlyPromotableShardsForStateless(
-            operationRouting.getShards(
-
+        ShardIterator iterator = clusterService.operationRouting()
+            .getShards(
                 project,
-
                 request.concreteIndex(),
-
                 request.request().id(),
-
                 request.request().routing(),
-
                 request.request().preference()
-
-            )
-        );
+            );
+        if (iterator == null) {
+            return null;
```

Comment:
So it will execute on the indexing shard in that case?

Reply:
I think that it'll execute the request in the receiving node. I think that we should return an empty iterator instead.